app/models/concerns/audit_extensions.rb

```diff
@@ -33,6 +51,70 @@ def ensure_taxonomies_not_escalated
     # Audits should never execute what Taxonomix#set_current_taxonomy does
     def set_current_taxonomy
     end
+
+    class << self
+      def location_taxed_auditables
+        select_auditable_types do |model|
+          has_any_association_to?(:location, model) &&
+          !has_any_association_to?(:organization, model)
+        end
+      end
+
+      def organization_taxed_auditables
+        select_auditable_types do |model|
+          has_any_association_to?(:organization, model) &&
+          !has_any_association_to?(:location, model)
+        end
+      end
+
+      def fully_taxable_auditable_types
+        select_auditable_types do |model|
+          [:location, :organization].map do |taxable|
+            Taxonomy.enabled?(taxable) && has_any_association_to?(taxable, model)
+          end.all?
+        end
+      end
+
+      def untaxable_auditables
+        select_auditable_types do |model|
+          !has_taxonomix?(model)
+          !has_any_association_to?(:organization, model) &&
+          !has_any_association_to?(:location, model)
+        end
+      end
+
+      def all_auditable_types
+        unscoped.distinct.pluck(:auditable_type).map do |auditable_type|
+          begin
+            auditable_type.constantize
+          rescue NameError
+          end
+        end.compact
+      end
+
+      def select_auditable_types(&block)
+        all_auditable_types.select(&block)
```

Comment:
TBH I don't think this removes much of duplicity, it makes the code less readable at the same time, not strong opinion but I'd prefer removing this